generators/git/index.js

```diff
@@ -22,13 +28,34 @@ module.exports = generators.Base.extend({
       this.templatePath('gitignore'),
       this.destinationPath('.gitignore')
     );
+
+    var done = this.async();
+
+    originUrl(this.destinationPath(), function (err, url) {
+      this.originUrl = url;
+      done();
+    }.bind(this));
+
+  },
+
+  writing: function () {
+    this.pkg = this.fs.readJSON(this.destinationPath('package.json'), {});
+    if (!this.pkg.repository) {
```

Comment:
To prevent nesting so deep, assign a variable like so:

```js
if (this.originUrl) {
  var repository = this.originUrl;
} else {
  //...
}

this.pkg.repository = this.pkg.repository || repository;
```

Or a similar logic. There's no need to nest to code so deep (mainly because if makes it hard to understand what is going on)